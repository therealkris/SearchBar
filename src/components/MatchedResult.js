// @flow
import * as React from "react";

type ResultProps = {
  term: string,
  result: string,
  handleClick: Function
};

export default function MatchedResult(props: ResultProps) {
  const { term, result, handleClick } = props;
  const match = result.replace(
    new RegExp(term, "gi"),
    "<span data-testid='matcher' style='color: #b9b9b9'>$&</span>"
  );

  const clicked = () => {
    return handleClick(result);
  };

  return <li dangerouslySetInnerHTML={{ __html: match }} onClick={clicked} />;
}
