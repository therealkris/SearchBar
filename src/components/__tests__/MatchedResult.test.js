import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";

import MatchedResult from "../MatchedResult";

test("renders correctly", () => {
  const { container, getByText, getByTestId } = render(
    <MatchedResult term={"re"} result={"Record"} handleClick={() => null} />
  );
  expect(container).toMatchSnapshot();
  expect(getByText(/cord/i)).toBeInTheDocument();
  expect(getByTestId("matcher")).toBeInTheDocument();
});
