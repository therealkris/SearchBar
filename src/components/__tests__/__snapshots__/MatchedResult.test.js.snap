// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <li>
    <span
      data-testid="matcher"
      style="color: #b9b9b9"
    >
      Re
    </span>
    cord
  </li>
</div>
`;
